core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -3801,13 +3806,17 @@ private static boolean all(SqlCall call) {
   protected RelNode convertInsert(SqlInsert call) {
     RelOptTable targetTable = getTargetTable(call);
 
+    callStack.push(call);
+
     final RelDataType targetRowType =
         validator().getValidatedNodeType(call);
     assert targetRowType != null;
     RelNode sourceRel =
         convertQueryRecursive(call.getSource(), true, targetRowType).project();
     RelNode massagedRel = convertColumnList(call, sourceRel);
 
+    callStack.pop();
```

Comment:
Why callStack only used in convertInsert,will the other convertXXX methods use callStack?

Such as convertMerge method.

Reply:
Merge sql syntax will also transform into SqlInsert operand and since i use it only for insertion - no other places are covered by now, thus no need to append it.

Reply:
Makes sense, but I’d feel more confident if the test also tested a MERGE statement.

Reply:
EnumerableTableModify#implement is not implemented for MERGE, throws exception (
```
    switch (getOperation()) {
    case INSERT:
      method = BuiltInMethod.INTO.method;
      break;
    case DELETE:
      method = BuiltInMethod.REMOVE_ALL.method;
      break;
    default:
      throw new AssertionError(getOperation());
```